README.md

```diff
@@ -38,174 +38,117 @@ Function        Set-CloudEventJsonData                             0.2.0      Cl
-   'key1' = 'value1'
-   'key2' = @{
-      'key3' = 'value3'
-   }
+$cloudEvent | Set-CloudEventJsonData -Data @{
+    'Foo' = 'Hello'
+    'Bar' = 'World'
 }
+
+
+DataContentType : application/json
+Data            : {
+                    "Bar": "World",
+                    "Foo": "Hello"
+                  }
+Id              : ac9b12d9-ae45-4654-a4d7-42bbf0d9816d
+DataSchema      :
+Source          : mailto:cncf-wg-serverless@lists.cncf.io
+SpecVersion     : V1_0
+Subject         :
+Time            : 4/26/2021 9:00:45 AM
+Type            : com.example.object.deleted.v2
 ```
 
-### Add **XML Data** to a CloudEvent object
+### Set **XML Data** to a CloudEvent object
 ```powershell
-$cloudEvent | Add-CloudEventXmlData -Data @{
-   'key1' = @{
-      'key2' = 'value'
-   }
+$cloudEvent | Set-CloudEventXmlData -Data @{
+    'xml' = @{
+        'Foo' = 'Hello'
+        'Bar' = 'World'
+    }    
 } `
 -AttributesKeysInElementAttributes $true
-```
-`AttributesKeysInElementAttributes` specifies how to format the XML. If `true` and the input Data hashtable has pairs of 'Attributes', 'Value' keys creates XML element with attributes, otherwise each key is formatted as XML element.<br/>
-If `true`
-```powershell
-   @{'root' = @{'Attributes' = @{'att1' = 'true'}; 'Value' = 'val-1'}}
-```
-is formatted as
-```xml
-<root att1="true">val-1</root>
-```
-If `false`
-```powershell
-@{'root' = @{'Attributes' = @{'att1' = 'true'}; 'Value' = 'val-1'}}
-```
-is formatted as
-```xml
-<root><Attributes><att1>true</att1></Attributes><Value>val-1</Value></root>
-```
 
+
+DataContentType : application/xml
+Data            : <xml><Bar>World</Bar><Foo>Hello</Foo></xml>
+Id              : ac9b12d9-ae45-4654-a4d7-42bbf0d9816d
+DataSchema      :
+Source          : mailto:cncf-wg-serverless@lists.cncf.io
+SpecVersion     : V1_0
+Subject         :
+Time            : 4/26/2021 9:00:45 AM
+Type            : com.example.object.deleted.v2
+```
 #### Add Custom Format Data to a CloudEvent object
 ```powershell
-$cloudEvent | Add-CloudEventData -DataContentType 'application/text' -Data 'wow'
+$cloudEvent | Set-CloudEventData -DataContentType 'application/text' -Data 'Hello World!'
 ```
 
 ### Convert a CloudEvent object to an HTTP message in **Binary** or **Structured** content mode
 ```powershell
+# Format binary cloud event HTTP message
 $cloudEventBinaryHttpMessage = $cloudEvent | ConvertTo-HttpMessage -ContentMode Binary
+
+# Format structured cloud event HTTP message
 $cloudEventStructuredHttpMessage = $cloudEvent | ConvertTo-HttpMessage -ContentMode Structured
 ```
 
 ### Send CloudEvent object to HTTP server
 ```powershell
-Invoke-WebRequest -Method POST -Uri 'http://my.cloudevents.server/' -Headers $cloudEventBinaryHttpMessage.Headers -Body $cloudEventBinaryHttpMessage.Body
+Invoke-WebRequest -Method POST -Uri 'http://my.cloudevents.server/' -Headers $cloudEventStructuredHttpMessage.Headers -Body $cloudEventStructuredHttpMessage.Body
```

Comment:
Since the example uses a Structured CloudEvent, are Headers actually used at all in this example? What is contained in the Headers of a structured CE? Just curious :)

Do we need to show two examples, e.g. using Headers+Body for binary mode?

Reply:
Headers follow the https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#3-http-message-mapping. In structure mode the ContentType header is `application/cloudevents`

>Do we need to show two examples, e.g. using Headers+Body for binary mode?
Not sure. I can leave only the structure mode?

Reply:
thx!